lib/publiccloud/provider.pm

```diff
@@ -248,6 +249,42 @@ sub destroy_instance {
     $self->ipa(cleanup => 1, running_instance_id => $instance->instance_id);
 }
 
+sub _get_config {
+    my ($self) = @_;
+
+    unless ($self->{rest_config}) {
+        my $config_file = $ENV{OPENQA_CONFIG} || '/etc/openqa';
+        $config_file .= '/publiccloud.ini';
+        my $cfg = Config::IniFiles->new(-file => $config_file)
+          or die("Failed to load $config_file");
+        $self->{rest_config} = {
+            url => $cfg->val('global', 'url'),
+            ssl_insecure => $cfg->val('global', 'ssl_insecure', 0),
+        };
+    }
+    return $self->{rest_config};
+}
+
+sub do_rest_api {
+    my ($self, $path, %args) = @_;
+    my $method = $args{method} // 'get';
+    my $ua     = Mojo::UserAgent->new;
+    my $config = $self->_get_config();
+    my $res;
+
+    $ua->insecure($config->{ssl_insecure});
+    for my $cnt (1 .. 3) {
+        $res = $ua->$method($config->{url} . $path)->result;
```

Comment:
yeah we really need to fail in `_get_config` method when url is not exists in config. We could really confusing error here from `$ua` in case `$config->{url}` is undef ...